rstudio/shiny
# Add domain context to session$onSessionEnded callbacks

This PR changes the callback execution context from "_some_ session ended" to "**this** session ended".

### Test File

`./onSessionEnded.R`
```r
library(shiny)

ui <- fluidPage(
  HTML("
  Reprex:<br>
  * Close window (if it doesn't auto close)<br>
  * Verify session token exists in console<br>
  "),
  tags$script("setTimeout(window.close, 3000);")
)

defined_with_no_session_context <- function() {
  message("Ended! Domain length: ", length(getDefaultReactiveDomain()))
  message("Ended! Token: ", getDefaultReactiveDomain()$token)
}

server <- function(input, output, session) {
  session$onSessionEnded(defined_with_no_session_context)
}

shinyApp(ui, server)
```


### Before PR

The `getDefaultReactiveDomain()` object does not exist.

```r
shiny::shinyAppFile("onSessionEnded.R")
#>
#> Listening on http://127.0.0.1:6026
#> Ended! Domain length: 0
#> Ended! Token: 
```

### After PR

The `getDefaultReactiveDomain()` object exists.

```r
shiny::shinyAppFile("onSessionEnded.R")
#>
#> Listening on http://127.0.0.1:3469
#> Ended! Domain length: 84
#> Ended! Token: bf248258f2896ba8e1ad08debf2b73c9
```

----------------------------

cc @trestletech @alandipert
